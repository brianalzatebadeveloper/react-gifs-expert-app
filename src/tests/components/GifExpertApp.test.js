import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GitExpertApp';

describe('Pruebas del componente <GifExpertApp />', () => {

  const wrapper = shallow( <GifExpertApp /> );

  test('Debe mostrase correctamente', () => { 
    
    expect( wrapper ).toMatchSnapshot();

  });

  test('debe mostrar una lista de categorias', () => {

    const categories = ['The rock','john cena'];
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories } /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );

  });

});