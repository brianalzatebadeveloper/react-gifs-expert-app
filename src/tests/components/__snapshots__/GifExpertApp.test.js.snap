// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifExpertApp /> Debe mostrase correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas del componente <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="The rock"
      key="The rock"
    />
    <GifGrid
      category="john cena"
      key="john cena"
    />
  </ol>
</Fragment>
`;
